qiskit/visualization/gate_map.py

```diff
@@ -745,6 +754,49 @@ def plot_coupling_map(
     return None
 
 
+def _set_default_coloring(qubit_color, line_color, num_qubits, coupling_map):
+    if qubit_color is None:
+        qubit_color = ["#648fff"] * num_qubits
+    if line_color is None:
+        line_color = ["#648fff"] * len(coupling_map) if coupling_map else []
+    return qubit_color, line_color
+
+
+def _color_faulty_backend(
+    backend, qubit_color, line_color, faulty_color="#f24b4b", disabled_color="#b59696"
+):
+    """Changes in place the colors in qubit_color and line_color with faulty_color for the gates/qubits
+    that are not operational and disabled_color with the gates/qubits that are going to be disabled
+    as a consequence"""
+    faulty_qubits = backend.properties().faulty_qubits()
+    faulty_gates = backend.properties().faulty_gates()
+    faulty_edges = [gates.qubits for gates in backend.properties().faulty_gates()]
+
+    if not (faulty_qubits or faulty_gates):
+        return
+
+    cmap = backend.configuration().coupling_map
+    n_qubits = backend.configuration().n_qubits
+
+    from qiskit.compiler.transpiler import (  # pylint: disable=cyclic-import
+        _connected_working_qubits,
+    )
```

Comment:
Instead of doing this via a private function in `qiskit.compiler.transpiler` I think we should put this in a more common place. I think probable as a private method on `BackendConfiguration`, `BackendProperties`, or `BackendV1` itself (I'd prefer the model objects though). It seems like a weird dependency tree to have the visualization import this from `transpile()` even at runtime.